src/index.js

```diff
@@ -198,8 +198,14 @@ class MixpanelTool {
     }
   }
 
+  base64regex = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/;
+  
   getProperties(dataParam) {
-    return JSON.parse(atob(decodeURIComponent(dataParam)));
+    let dataStr = decodeURIComponent(dataParam);
+    if(base64regex.test(dataStr)) {
```

Comment:
I have an error here: base64regex is undefined. I have placed the definition of the var **inside** the method `getProperties` and it fixes the app.

Reply:
Whoops, sorry about that. I would recommend keeping the regex declared at the class level so it can be reused rather than recreated for every request
```suggestion
    if(this.base64regex.test(dataStr)) {
```